internal/controllers/topology/cluster/patches/variables/variables_test.go

```diff
@@ -221,3 +416,12 @@ func TestMachineDeployment(t *testing.T) {
 func toJSON(value string) apiextensionsv1.JSON {
 	return apiextensionsv1.JSON{Raw: []byte(value)}
 }
+
+// toJSONCompact is used to be able to write JSON values in a readable manner.
+func toJSONCompact(value string) apiextensionsv1.JSON {
+	var compactValue bytes.Buffer
+	if err := json.Compact(&compactValue, []byte(value)); err != nil {
+		panic(err)
+	}
+	return apiextensionsv1.JSON{Raw: compactValue.Bytes()}
+}
```

Comment:
can we do something like yaml.Raw in util/yaml, thus allowing a better indentation/readability in tests?

Reply:
Agreed - I just ported over #6171 to this function and it's not a great experience (still better than what was here before though!). We could just have a method that marshals the structs we want to json, right?

Reply:
Its about using yaml instead of json right?

Reply:
Sorry - I'm confusing the issue :smile: I just think this solution doesn't make tests easier to write. Yaml would maybe be an improvement - but I think it would be easiest if we could just declare the struct we want and then produce it as json.

Reply:
For me it is about readability. So we can easier see what we're testing

Reply:
If think it's beneficial to write the values in plain YAML or JSON. Eventually this will be what we sent over the wire to external patch implementations.

For example, I would never have found out that I have to use pointers for `MachineDeploymentBootstrapBuiltins` and `MachineDeploymentInfrastructureRefBuiltins` because otherwise we would have ended up with empty objects in the marshalled output (even with omitempty).

I also don't really think that writing this as a Go struct is easier. It's more typesafe but plain JSON is easier to write. Also this way we would detect errors in our JSON marshalling, e.g. as in the example I mentioned or when the JSON tags are incorrect. I see that as double-checking that the variables actually result in what we want. 

When looking at the usage of variables in JSON patches, they will eventually be used as JSON, which is why I would prefer keeping plain JSON here.

Reply:
Talked to Fabrizio a bit, it's about indentation not YAML or JSON. Will fix :)

Reply:
Updated the test. the json.Compact func I'm using is already dropping the indentation. So I just reformatted the file.

Note: `toJSONCompact` returns a `apiextensionsv1.JSON` which is why I couldn't just use `yaml.Raw`